acceptance_tests/utilities/pubsub_helper.py

```diff
@@ -53,10 +56,47 @@ def _ack_all_on_subscription(subscriber, subscription_path):
         return
 
     ack_ids = [message.ack_id for message in response.received_messages]
-
     if ack_ids:
         subscriber.acknowledge(subscription_path, ack_ids)
 
     # It's possible (though unlikely) that they could be > max_messages on the topic so keep deleting till empty
     if len(response.received_messages) == max_messages_per_attempt:
         _ack_all_on_subscription(subscriber, subscription_path)
+
+
+def _pull_exact_number_of_messages(subscriber, subscription_path, expected_msg_count, timeout):
+    # Synchronously pull messages one at at time until we either hit the expected number or the timeout passes.
+    messages = []
+    deadline = time.time() + timeout
+
+    while len(messages) < expected_msg_count and not time.time() > deadline:
```

Comment:
this loop will be pretty CPU intensive won't it? Shouldn't it have a sleep of a few milliseconds, just to save maxing the CPU?